hadoop-ssm-project/src/main/java/org/apache/hadoop/ssm/protocolPB/ClientSSMProtocolClientSideTranslatorPB.java

```diff
@@ -38,18 +39,23 @@
 import java.util.ArrayList;
 import java.util.List;
 
-public class ClientSSMProtocolClientSideTranslatorPB implements ClientSSMProtocol {
+public class ClientSSMProtocolClientSideTranslatorPB implements java.io.Closeable,ClientSSMProtocol {
   final private ClientSSMProtocolPB rpcProxy;
 
   public ClientSSMProtocolClientSideTranslatorPB(ClientSSMProtocolPB proxy) {
     this.rpcProxy = proxy;
   }
 
   @Override
-  public SSMServiceState getServiceState() {
+  public SSMServiceState getServiceState() throws IOException {
     GetServiceStateRequestProto req =
         GetServiceStateRequestProto.newBuilder().build();
-    int state = rpcProxy.getServiceState(null, req).getState();
+    int state = 0;
+    try {
+      state = rpcProxy.getServiceState(null, req).getState();
+    } catch (ServiceException e) {
+      throw PBHelper.getRemoteException(e);
+    }
     return SSMServiceState.fromValue(state);
```

Comment:
move this line into try block, it does not need to put 'state' definition outside of the try block